server/src/main/java/org/elasticsearch/action/admin/cluster/node/hotthreads/NodesHotThreadsResponse.java

```diff
@@ -27,6 +34,17 @@ public NodesHotThreadsResponse(ClusterName clusterName, List<NodeHotThreads> nod
         super(clusterName, nodes, failures);
     }
 
+    public Iterator<CheckedConsumer<java.io.Writer, IOException>> getTextChunks() {
```

Comment:
Nit: I would have put this as a static private method inside RestNodesHotThreadsAction, since it is the only usage and it uses only public methods (and the LinesIterator class too), because I like to keep data/transfer objects as bare as possible, but this is just a matter of preference.

Reply:
We're pretty big fans of having rendering logic implemented within DTOs (see for instance `ToXContent` and `ChunkedToXContent`) and not within the REST layer (see for instance the lack of unit tests of pretty much all Rest*Action classes).

Reply:
Good to know, I will follow this convention then. Thanks for pointing it out!